applications/FluidDynamicsApplication/custom_processes/calulate_levelset_consistent_nodal_gradient_process.cpp

```diff
@@ -0,0 +1,148 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Mohammad R. Hashemi
+//
+
+// Project includes
+#include "includes/define.h"
+#include "utilities/parallel_utilities.h"
+
+// Application includes
+#include "fluid_dynamics_application_variables.h"
+#include "calulate_levelset_consistent_nodal_gradient_process.h"
+
+
+namespace Kratos
+{
+
+/* Public functions *******************************************************/
+
+/// constructor
+CalulateLevelsetConsistentNodalGradientProcess::CalulateLevelsetConsistentNodalGradientProcess(
+        ModelPart& rModelPart)
+    : Process(), mrModelPart(rModelPart) {}
+
+/// Constructor with Kratos parameters.
+CalulateLevelsetConsistentNodalGradientProcess::CalulateLevelsetConsistentNodalGradientProcess(
+    ModelPart& rModelPart,
+    Parameters Parameters)
+    : CalulateLevelsetConsistentNodalGradientProcess(
+        rModelPart
+    ){}
+
+/// Constructor with Kratos model
+CalulateLevelsetConsistentNodalGradientProcess::CalulateLevelsetConsistentNodalGradientProcess(
+    Model& rModel,
+    Parameters Parameters)
+    : CalulateLevelsetConsistentNodalGradientProcess(
+        rModel.GetModelPart(Parameters["model_part_name"].GetString())
+    ){}
+
+void CalulateLevelsetConsistentNodalGradientProcess::Execute(){
+
+    KRATOS_TRY;
+
+    const auto zero_vector = ZeroVector(3);
+    // Set to zero
+    block_for_each(mrModelPart.Nodes(), [&](Node<3>& rNode){
+        rNode.SetValue(NODAL_AREA, 0.0);
+        rNode.SetValue(PRESSURE_GRADIENT, zero_vector);
+    });
+
+    // Current domain size
+    const unsigned int num_dim = mrModelPart.GetProcessInfo()[DOMAIN_SIZE];
+    const unsigned int num_nodes = num_dim + 1; //For tetrahedra and triangles
+
+    // Auxiliar containers
+    double detJ0 = 0.0;
+    Matrix DN_DX, J0, InvJ0;
+    Vector N(num_nodes), pressures(num_nodes), distances(num_nodes), grad(num_dim);
+
+    // First element iterator
+    const auto it_element_begin = mrModelPart.ElementsBegin();
+
+    const int elements_number = mrModelPart.Elements().size();
+
+    // Iterate over the elements
+    #pragma omp parallel for firstprivate(DN_DX, N, J0, InvJ0, detJ0, pressures, distances, grad)
+    for(int i_elem = 0; i_elem < elements_number; ++i_elem) {
+
+        const auto it_elem = it_element_begin + i_elem;
+        auto& r_geometry = it_elem->GetGeometry();
```

Comment:
```suggestion
        const auto& r_geometry = it_elem->GetGeometry();
```
If I'm not wrong all the functions of the geometry that you use have a `const` version.

Reply:
Seems so. I change them all.

Reply:
Was not possible due to following changes.

Reply:
OK then!. Resolving.